neuromation/cli/job.py

```diff
@@ -1127,3 +1176,54 @@ def calc_statuses(status: Sequence[str], all: bool) -> Set[JobStatus]:
                 return parse_columns(format_str)
         return COLUMNS
     return format
+
+
+async def calc_life_span(client: Client, value: Optional[str]) -> timedelta:
+    result = (
+        _parse_timedelta(value)
+        if value is not None
+        else await calc_default_life_span(client)
+    )
+    seconds = result.total_seconds()
+    if seconds == 0:
+        return timedelta.max
+    elif seconds < 0:
+        raise click.UsageError(f"Job life span must be non-negative, got: '{result}'")
```

Comment:
technically, our input parser does not allow negative values, so they can come only from the config. I understand, it's better to check it during user config validation, but it's quicker to do it here, after parsing.
1. Should I add more validation code to `_validate_user_config()`, similarly as it is with `_validate_alias()`?
2. Why don't we use `trafaret`? Doesn't it have enough expressive power to validate aliases, does it?

Reply:
> technically, our input parser does not allow negative values, so they can come only from the config. I understand, it's better to check it during user config validation, but it's quicker to do it here, after parsing.
> 
> 1. Should I add more validation code to `_validate_user_config()`, similarly as it is with `_validate_alias()`?

Please keep as is until we implement something like `neuro config validate` command.

> 2. Why don't we use `trafaret`? Doesn't it have enough expressive power to validate aliases, does it?

Trafaret provides machine-readable error messages which should be translated to human-readable texts before printing. It makes `trafaret` virtually useless for our needs.

Reply:
Why not to use [`extract_error`](https://trafaret.readthedocs.io/en/latest/api/trafaret.html#trafaret.extract_error)? It might be easier to use a package that was designed to write custom validators, rather than to implement our own from scratch.